core/src/nl/tudelft/watchdog/core/logic/ui/listeners/staticanalysis/CoreMarkupModelListener.java

```diff
@@ -0,0 +1,66 @@
+package nl.tudelft.watchdog.core.logic.ui.listeners.staticanalysis;
+
+import nl.tudelft.watchdog.core.logic.document.Document;
+import nl.tudelft.watchdog.core.logic.event.TrackingEventManager;
+import nl.tudelft.watchdog.core.logic.event.eventtypes.TrackingEventType;
+import nl.tudelft.watchdog.core.logic.event.eventtypes.staticanalysis.StaticAnalysisWarningEvent;
+import org.joda.time.DateTime;
+
+import java.util.stream.Stream;
+
+/**
+ * Base class for a MarkupModelListener intended to listen to static analysis warning changes.
+ * It can processes {@link Stream} of {@link Warning} and generates the corresponding
+ * {@link StaticAnalysisWarningEvent} which is added to the {@link #trackingEventManager}.
+ */
+public class CoreMarkupModelListener {
```

Comment:
I prefer to keep it `Core`, for easier searching in my IDE. Auto-completers like that better, rather than typing out the whole name and then having to add `Base` to it at the end.